python/pyarrow/array.pxi

```diff
@@ -3825,7 +3825,7 @@ cdef class StructArray(Array):
         return [pyarrow_wrap_array(arr) for arr in arrays]
 
     @staticmethod
-    def from_arrays(arrays, names=None, fields=None, mask=None,
+    def from_arrays(arrays, names=None, fields=None, structtype=None, mask=None,
```

Comment:
as those are not keyword only arguments this is changing the API order, we should add it either as a keyword only argument or as the last argument of the function.

Reply:
You can also name the argument `type` instead of `structtype` to be consistent with other `from_arrays` methods